docs/rules/prefer-find-by.md

```diff
@@ -0,0 +1,43 @@
+# Use `find*` query methods to wait for elements instead of waitFor (prefer-find-by)
+
+TBD
+
+## Rule details
+
+This rule aims to use `findBy*` or `findAllBy*` queries to wait for elements, rather than using `waitFor`, or the deprecated methods `waitForElement` and `wait`
+
+Examples of **incorrect** code for this rule
+
+```js
+const submitButton = await waitFor(() =>
+  screen.getByRole('button', { name: /submit/i })
+);
+
+const submitButton = await waitFor(() =>
+  screen.getAllTestId('button', { name: /submit/i })
+);
+
+const submitButton = await waitFor(() =>
+  queryByLabel('button', { name: /submit/i })
+);
+
+const submitButton = await waitFor(() =>
+  queryAllByText('button', { name: /submit/i })
+);
+```
+
+Examples of **correct** code for this rule:
+
+```js
+const submitButton = await findByText('foo');
+
+const submitButton = await screen.findAllByRole('table');
+```
+
+## When Not To Use It
+
+TBD
+
+## Further Reading
+
+TBD
```

Comment:
I would link to [findBy query doc](https://testing-library.com/docs/dom-testing-library/api-queries#findby) here, as it mentions that `findBy` is a shortcut for `waitFor` + `getBy`. I would use the text from that callout as the main inspiration for this url in the description at the top.